From 9248a69bc147c3a43b1af29b95f5b0b67c7a7fcb Mon Sep 17 00:00:00 2001
From: Frederik Schwan <frederik@tty42.de>
Date: Mon, 10 Jun 2024 10:23:11 +0200
Subject: [PATCH] Pin Java Version

Inject JAVA_HOME and JAVA variables into scripts that end up in /usr/bin
to also pin the java version for the cli tools that otherwise would use
the default system JAVA_HOME and JAVA var.# Please enter the commit message for your changes. Lines starting
---
 integration/client-cli/admin-cli/src/main/bin/kcadm.sh | 1 +
 integration/client-cli/admin-cli/src/main/bin/kcreg.sh | 1 +
 2 files changed, 2 insertions(+)

diff --git a/integration/client-cli/admin-cli/src/main/bin/kcadm.sh b/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
index fa0e7f86a5..a08299e498 100755
--- a/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
+++ b/integration/client-cli/admin-cli/src/main/bin/kcadm.sh
@@ -19,6 +19,7 @@ esac
 RESOLVED_NAME="${RESOLVED_NAME:-"$0"}"
 
 DIRNAME="$(dirname "$RESOLVED_NAME")"
+. "$DIRNAME/common.sh"
 
 if [ -z "$JAVA" ]; then
     if [ -n "$JAVA_HOME" ]; then
diff --git a/integration/client-cli/admin-cli/src/main/bin/kcreg.sh b/integration/client-cli/admin-cli/src/main/bin/kcreg.sh
index be65bcadb1..ef135760e4 100755
--- a/integration/client-cli/admin-cli/src/main/bin/kcreg.sh
+++ b/integration/client-cli/admin-cli/src/main/bin/kcreg.sh
@@ -27,4 +27,5 @@ if [ -z "$JAVA" ]; then
 fi
 
 DIRNAME="$(dirname "$RESOLVED_NAME")"
+. "$DIRNAME/common.sh"
 exec "$JAVA" $KC_OPTS -cp $DIRNAME/client/keycloak-admin-cli-${project.version}.jar --add-opens=java.base/java.security=ALL-UNNAMED -Dkc.lib.dir=$DIRNAME/client/lib org.keycloak.client.registration.cli.KcRegMain "$@"
-- 
2.45.2

